sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVExprUtils.scala

```diff
@@ -79,4 +81,39 @@ object CSVExprUtils {
         throw new IllegalArgumentException(s"Delimiter cannot be more than one character: $str")
     }
   }
+
+  /**
+   * Helper method that converts string representation of a character sequence to actual
+   * delimiter characters. The input is processed in "chunks", and each chunk is converted
+   * by calling [[CSVExprUtils.toChar()]].  A chunk is either:
+   * <ul>
+   *   <li>a backslash followed by another character</li>
+   *   <li>a non-backslash character by itself</li>
+   * </ul>
+   * , in that order of precedence. The result of the converting all chunks is returned as
```

Comment:
You might throw in an example here. The idea of a chunk is just to account for expressing a tab as `\t`, right? That kind of example could clarify this. Really the delimiter is just a string whose backslash escapes have been resolved in the same way that a Scala / Java string does (right?)

This documentation is fine here but is even more important in the docs for what the CSV reader accepts.

Reply:
I added some examples to the Scaladoc, and also a new test suite that shows various inputs and outputs.  And yes, your description is accurate.  This method is a bit analogous to `StringEscapeUtils#unescapeJava` from Apache `commons-text`.

Regarding the docs, I'm happy to take a look if you point out where that should be.

Reply:
Correction: created new test in existing `CSVExprUtilsSuite`.

Reply:
Take a look at `DataFrameReader.scala` and `def csv(paths: String*)`. There's a big chunk of API doc that describes the options. This should be updated, along with the equivalent in Pyspark, and SparkR if it exists (I forget)

Reply:
I updated what I understood to be the correct spots in the Python files, with the same new text (happy to reword as needed).  I searched a few different ways for the equivalent in the `.R` files and couldn't find anything.